hw2-turtle-graphics/turtle-stretch1.js

```diff
@@ -0,0 +1,149 @@
+                    this.y++
+                    this.allpoints.push([this.x,this.y])
+
+                } else if (this.direction === "west") {
+                    this.x--
+                    this.allpoints.push([this.x,this.y])
+                }
+            }   
+
+            //getting the highest value for the x axis and y axis, will be used to create the grid later
+
+            if(this.x > this.maxX){
+                this.maxX = this.x 
+            }  else if (this.y > this.maxY){
+                this.maxY = this.y
+            }
+
+        return this;
+        } 
+
+        right() {
+        //checks the current direction to turn the turn "right" from the current direction
+            if (this.direction === "north") {
+                this.direction = "east"
+
+            } else if (this.direction == "east") {
+                this.direction = "south"
+
+            } else if (this.direction === "south") {
+                this.direction = "west"
+
+            } else if (this.direction === "west") {
+                this.direction = "north"
+            }
+        return this;
+        }
+
+        left() {
+        //checks the current direction, turns the turtle left
+
+            if (this.direction == "north") {
+                this.direction = "west"
+
+            } else if (this.direction == "east") {
+                this.direction = "north"
+
+            } else if (this.direction == "south") {
+                this.direction = "east"
+
+            } else if (this.direction == "west") {
+                this.direction = "south"
+            }
+            return this;
+        }
+
+        allPoints(){
+        return(this.allpoints)
+        } 
+        
+        print(){
+            
+            const final = [];
+            const slicedpoints = this.allpoints;
+
+            for (let i=0 ; i<this.maxX+2; i++){         //iterates through the width of the grid            
+                const x = []
+                for(let j=0; j<this.maxY+2; j++){  //iterates through the length of the grid
+                    const gridPath = slicedpoints.find((item)=>{
+                    return item.join(",")===`${j},${i}` 
+                    })
+                    if(gridPath) x.push("■") // if the box is crossed by the turtle-- fills the box, pushes to x array
+                    else x.push("□") // if not, the box will be empty- push to x array
+                }
+        
+                final.push(x.join("")) //pushes x arrays to final array
+                }
+                //joins and prints the final arrays-- printing them on their own lines to print the grid.
+                console.log("--BEGIN LOG")
+                console.log(final.join("\n"))
+                console.log("--END LOG")
+        }  
+ }
+
+ function stretch() {
+    // taking the arguments, splitting them into an array separated by the dashes
+    
+    let commands = process.argv.slice(2);
```

Comment:
you can also do `process.argv.slice(2)?. toString()` but what you have also works.